### PR TITLE
Added test_commitlog_replay_on_startup

### PR DESCRIPTION
**Pull request by pauloricardomg**
During #CASSANDRA-10051 I needed to investigate a difference in behavior of commit log replay between Windows and Linux, and didn't find any dtest for commit log replay, which is a critical feature of cassandra.

This dtest basically inserts a single row, shuts down the node without flushing, and verify if the inserted row is present in next cluster startup due to commit log replay.

### CODE REVIEW

**Comment by ptnapoleon** [comment]
Do none of the commitlog tests really cover this?

File: commitlog_test.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (ptnapoleon) [inline commitlog_test.py]: Why the new line?

**Comment by ptnapoleon** [comment]
+1

File: commitlog_test.py (hunk 1 above)
Comment (pauloricardomg) [inline commitlog_test.py]: mostly to separate testing steps.. but not a big deal, you can remove if you want :)

**Comment by pauloricardomg** [comment]
surprisingly I didn't find any test that cover this.. at least explicitly. maybe some other test covers it indirectly, or maybe I just missed it.

**Comment by ptnapoleon** [comment]
Well, that's upsetting.

**Pull request merged**